lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -25,58 +28,306 @@
+    128: "accumulation or average",
+    64: "from UM 150m",
+    32: "scaled to UM resolution",
+    16: "averaged over multiple surface types",
+    8: "only observations used",
+    4: "smoothed",
+    2: "cold bias applied",
+    1: "warm bias applied",
+}
 
 
 class TranslationWarning(Warning):
     pass
 
 
-def name(cube, field):
-    """Set the cube's name from the field."""
+def is_missing(field, value):
+    """Returns True if value matches an "is-missing" number."""
+    return any(
+        np.isclose(value, [field.int_mdi, field.float32_mdi, NIMROD_DEFAULT])
+    )
 
-    cube.rename(field.title.strip())
+
+def name(cube, field):
+    """Set the cube's name from the field.
+    Modifies the Nimrod object title based on other meta-data in the
+    Nimrod field and known use cases.
+    """
+    if field.field_code == 12:
+        field.title = "air_pressure"
+    if field.field_code == 27:
+        field.title = "snow fraction"
+    if field.field_code == 28:
+        field.title = "snow probability"
+    if field.field_code == 29 and field.threshold_value >= 0.0:
+        field.title = "fog fraction"
+    if field.field_code == 58:
+        field.title = "temperature"
+    if field.field_code == 61:
+        field.title = "precipitation"
+    if field.field_code == 63:
+        field.title = "precipitation"
+    if field.field_code == 101:
+        field.title = "snow_melting_level_above_sea_level"
+    if field.field_code == 102:
+        field.title = "rain_melted_level_above_sea_level"
+    if field.field_code == 155:
+        field.title = "Visibility"
+    if field.field_code == 156:
+        field.title = "Worst visibility in grid point"
+    if field.field_code == 161 and field.threshold_value >= 0.0:
+        field.title = "minimum_cloud_base_above_threshold"
+    if field.field_code == 218:
+        field.title = "snowfall"
+    if field.field_code == 172:
+        field.title = "cloud_area_fraction_in_atmosphere"
+    if field.field_code == 421:
+        field.title = "precipitation type"
+    if field.field_code == 501:
+        field.title = "vector_wind_shear"
+        field.source = ""
+    if field.field_code == 508:
+        field.title = "low_level_jet_u_component"
+    if field.field_code == 509:
+        field.title = "low_level_jet_curvature"
+    if field.field_code == 514:
+        field.title = "low_level_jet_v_component"
+    if field.field_code == 804 and field.vertical_coord >= 0.0:
+        field.title = "wind speed"
+    if field.field_code == 806 and field.vertical_coord >= 0.0:
+        field.title = "wind direction"
+    if field.field_code == 817:
+        field.title = "wind_speed_of_gust"
+    if field.field_code == 821:
+        field.title = "Probabilistic Gust Risk Analysis from Observations"
+        field.source = "Nimrod pwind routine"
+    if field.source.strip() == "pwind":
+        field.source = "Nimrod pwind routine"
+
+    if getattr(field, "ensemble_member") == -98:
```

Comment:
I don't think there is any value in using `getattr` with a literal name, and no default value.
You can just replace with `field.ensemble_member`

This occurs in 3 places, all relating to 'ensemble_member'
 -- two here, and one in the `ensemble_member` routine.

Reply:
Agreed and changed.